### PR TITLE
[RELY-67] Update Node Logger

### PR DESCRIPTION
**Pull request by MFrizo**
# What does this PR do?

Previously at Jaiminho's codebase, it was necessary to implement a new version of the logging process to manage properly the debugging process and add updated versions of the libraries used for logging. Once this code is already running at Jaiminho for the last four months, we can ensure that it works properly.

The purpose of this PR is to move the code implemented in the Jaiminho [here](https://github.com/quintoandar/jaiminho/tree/master/app/helpers/logger) to this lib and allow the other applications to have access to these new features. In order to ease the bumping process, we maintain the same interface of the previous version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,13 +45,55 @@ logger.error(`Some error while processing cool object with id ${object.id}`, { e
 ```
 
 On the console it will be logged as a json:
+
 ```sh
-{"level":"info","message":"Some info about processing cool object with id 10","extra_data":{"extra":{"data":{"id":"11","someInfo":"someInfo"}}},"logger_name":"path/to/my/file.js","timestamp":"2018-12-19T18:15:57.078Z"}
+[info] Some info about processing cool object with id 11 [trace-id: undefined] {  extra: {    extra: {      data: {        id: 11,        someInfo: 'someInfo'      }    }  },  module: 'path/to/my/file.js',  timestamp: '2020-06-08T15:32:37.472Z'}
```
Comment (thiagoazeredo-quintoandar) [inline README.md]: Are you sure it is printinng trace-id undefined?

Comment (thiagoazeredo-quintoandar) [reply README.md]: Other, I think you did a config to expose the timestamp

File: README.md [hunk 2]
```diff
@@ -43,13 +45,55 @@ logger.error(`Some error while processing cool object with id ${object.id}`, { e
...
+  module: 'path/to/my/file.js',
+  timestamp: '2020-06-08T15:35:29.122Z'
+}
 ```
 
 And on Sentry the data on `extra` will be displayed under the field `Additional Data`.
 
+### Tracer
+
+If your project is using some tracer library, you can include the tracer Id of the instance running to the logger.
```
Comment (thiagoazeredo-quintoandar) [inline README.md]:
```suggestion
If your project is using the honeycomb tracer library, you can include the tracer Id of the instance running to the logger.
```

File: src/main.js [hunk 3]
```diff
@@ -1,43 +1,196 @@
...
+}
+
+const _consoleLog = console.log;
+const _consoleError = console.error;
+const _consoleWarn = console.warn;
+
+const sentryMessageToAvoid = 'Sentry Logger';
+const sentryMessageType = 'string';
+
+if (process.env.NODE_ENV !== 'test') {
```
Comment (thiagoazeredo-quintoandar) [inline src/main.js]: I would like to invert it...
```suggestion
if (process.env.NODE_ENV === 'test') {
    return;
 }
console.log.......    
```

File: README.md (hunk 1 above)
Comment (MFrizo) [reply README.md]: > Are you sure it is printinng trace-id undefined?

It's wrong actually, I'll update it.

Comment (MFrizo) [reply README.md]: > Other, I think you did a config to expose the timestamp

Yes, the `NODE_ENV` env var should be different of `production`. I'll add it to README too.

**Review by thiagoazeredo-quintoandar — APPROVED** [review]

**Review by evandrocarrenho — APPROVED** [review]

**Review by github-felipe-caputo — APPROVED** [review]

**Pull request merged**